src/IECoreImage/DisplayDriverServer.cpp

```diff
@@ -423,9 +437,39 @@ void DisplayDriverServer::Session::handleReadOpenParameters( const boost::system
 
 		const StringData *displayType = parameters->member<StringData>( "remoteDisplayType", true /* throw if missing */ );
 
+		const BoolData *mergeDriverData = parameters->member<BoolData>( "mergeDriver", false);
+        m_mergeSession = mergeDriverData && mergeDriverData->readable();
+
 		// create a displayDriver using the factory function.
-		m_displayDriver = DisplayDriver::create( displayType->readable(), displayWindow->readable(), dataWindow->readable(), channelNames->readable(), parameters );
+		if (!m_mergeSession)
+		{
+			m_displayDriver = DisplayDriver::create( displayType->readable(), displayWindow->readable(), dataWindow->readable(), channelNames->readable(), parameters );
+		}
+		else
+		{
+			m_mergeId = parameters->member<IntData>( "sessionId", true /* throw if missing */ )->readable();
 
+			// Check if merge ID in map, if not then create display driver and session count pair with merge ID.
+			if (const auto search = mergeMap.find(m_mergeId); search == mergeMap.end())
+			{
+				const IntData *sessionClientsData = parameters->member<IntData>( "sessionClients", true /* throw if missing */ );
```

Comment:
I assume this parameter tells us how many total clients there are going to be in this session, and it is the responsibility of the person setting up the clients to set it correctly?

I did wonder if we could remove the need for this by just incrementing the client count each time we add a client to the merge. But I suppose it is possible for one very slow machine (or a farm resource allocation problem) to mean that one client doesn't even start before all the others have finished.

Reply:
Yes it is the responsability of the client code to set the parameter up properly. 

And you're right I did think about an iterator somewhere, but it could be the case that all but one machine finishes, whilst one machine hasn't even started yet, which would cause the driver to close, and hence the catalogue image.